compiler/src/dotty/tools/dotc/typer/RefChecks.scala

```diff
@@ -99,10 +100,14 @@ object RefChecks {
           report.error(DoesNotConformToSelfType(category, cinfo.selfType, cls, otherSelf, relation, other.classSymbol),
             cls.sourcePos)
       }
-      for (parent <- cinfo.classParents)
+      val parents = cinfo.classParents
+      for (parent <- parents)
         checkSelfConforms(parent.classSymbol.asClass, "illegal inheritance", "parent")
       for (reqd <- cinfo.cls.givenSelfType.classSymbols)
         checkSelfConforms(reqd, "missing requirement", "required")
+
+      if !cls.is(Enum) && parents.exists(_.classSymbol == defn.JavaEnumClass) then
```

Comment:
Secondly, I believe It should be allowed for traits, e.g. we can do the following without issue:
```scala
trait T extends java.lang.Enum[T]

enum TEnum extends T {
  case Foo, Bar
}

object Test extends App {
  println(TEnum.Foo)
}
```

The parents will be deeply checked for any class that extends the trait that is not an enum